packages/monaco-editor-wrapper/src/editorAppBase.ts

```diff
@@ -223,3 +227,56 @@ export abstract class EditorAppBase {
 export const isVscodeApiEditorApp = (wrapperConfig: WrapperConfig) => {
     return wrapperConfig.editorAppConfig?.$type === 'vscodeApi';
 };
+
+export const isCodeUpdateRequired = (config: EditorAppBaseConfig, modelUpdate: ModelUpdate) => {
+    let updateRequired = modelUpdate.code !== undefined && modelUpdate.code !== config.code;
+    updateRequired = updateRequired || modelUpdate.codeOriginal !== config.codeOriginal;
+    return updateRequired ? ModelUpdateType.code : ModelUpdateType.none;
+};
+
+export const isModelUpdateRequired = (config: EditorAppBaseConfig, modelUpdate: ModelUpdate): ModelUpdateType => {
+    const codeUpdate = isCodeUpdateRequired(config, modelUpdate);
+    let updateRequired = modelUpdate.languageId !== config.languageId;
```

Comment:
A simple conditional branch here could work as well, avoiding the repeated assignment, and allowing short circuiting to take affect.